web/src/core/adapters/s3Client/s3Client.ts

```diff
@@ -331,24 +331,62 @@ export function createS3Client(
 
             const { awsS3Client } = await getAwsS3Client();
 
-            const bucketPolicyAndAllowedPrefix = await (async () => {
-                const { GetBucketPolicyCommand } = await import("@aws-sdk/client-s3");
+            const resolveBucketPolicy = async () => {
+                const { GetBucketPolicyCommand, S3ServiceException } = await import(
+                    "@aws-sdk/client-s3"
+                );
 
                 let sendResp: import("@aws-sdk/client-s3").GetBucketPolicyCommandOutput;
-
                 try {
                     sendResp = await awsS3Client.send(
-                        new GetBucketPolicyCommand({
-                            Bucket: bucketName
-                        })
+                        new GetBucketPolicyCommand({ Bucket: bucketName })
                     );
-                } catch {
-                    console.log("The error is ok, there is no bucket policy");
-                    return undefined;
-                }
+                } catch (error) {
+                    if (error instanceof S3ServiceException) {
+                        switch (error.$metadata?.httpStatusCode) {
+                            case 404:
+                                console.info(
+                                    "Bucket policy does not exist (404), it's ok."
+                                );
+                                return {
+                                    isBucketPolicyAvailable: true,
+                                    bucketPolicy: undefined,
+                                    allowedPrefix: []
+                                };
+                            case 403:
+                                console.info("Access denied to bucket policy (403).");
+                                return {
+                                    isBucketPolicyAvailable: false,
+                                    bucketPolicy: undefined,
+                                    allowedPrefix: []
+                                };
+                            default:
+                                console.error("An S3 error occurred:", error.message);
+                                return {
+                                    isBucketPolicyAvailable: false,
+                                    bucketPolicy: undefined,
+                                    allowedPrefix: []
+                                };
+                        }
+                    }
 
-                if (sendResp.Policy === undefined) {
-                    return undefined;
+                    console.error(
+                        "An unexpected error occurred when fetching bucket policy",
+                        error
+                    );
+                    return {
+                        isBucketPolicyAvailable: false,
+                        bucketPolicy: undefined,
+                        allowedPrefix: []
+                    };
```

Comment:
I'd rather see:  

```ts
try{
}catch(error){
    if( !(error instanceof S3ServiceException)){
        throw error;
    }
    switch(...
}
```

The reasoning is that we should be consistent with our error management policies.  
If we wheren't handling `S3ServiceException` we would let the exception stop the execution.  

As of now, any other type of exception that might be thrown here is unexpected and we should adress it.